ui/src/app/components/cost-predictor.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+
+import colors, { colorWithWhiteness } from 'app/styles/colors';
+import { formatUsd } from 'app/utils/numbers';
+import { isUsingFreeTierBillingAccount } from 'app/utils/workspace-utils';
+
+import { FlexRow } from './flex';
+import { StartStopRuntimeButton } from './runtime-configuration-panel/start-stop-runtime-button';
+import { styles } from './runtime-configuration-panel/styles';
+import { RuntimeCostEstimator } from './runtime-cost-estimator';
+import { Spinner } from './spinners';
+
+const CostInfo = ({
+  creatorFreeCreditsRemaining,
+  runtimeChanged,
+  analysisConfig,
+  currentUser,
+  workspace,
+}) => {
+  const remainingCredits =
+    creatorFreeCreditsRemaining === null ? (
+      <Spinner size={10} />
+    ) : (
+      formatUsd(creatorFreeCreditsRemaining)
+    );
+
+  return (
+    <FlexRow data-test-id='cost-estimator'>
+      <div
+        style={{
+          padding: '.495rem .75rem',
+          ...(runtimeChanged
+            ? {
+                backgroundColor: colorWithWhiteness(colors.warning, 0.9),
+              }
+            : {}),
+        }}
+      >
+        <RuntimeCostEstimator {...{ analysisConfig }} />
```

Comment:
If this handles more than just runtimes now, we should probably rename it

Reply:
Renamed it to EnvironmentCostEstimator